responses/tests/test_responses.py

```diff
@@ -2342,3 +2345,114 @@ def run():
+        """This example is present in README.rst"""
+
+        @responses.activate(registry=registries.OrderedRegistry)
+        def run():
+            url = "https://example.com"
+            rsp1 = responses.get(url, body="Error", status=500)
+            rsp2 = responses.get(url, body="Error", status=500)
+            rsp3 = responses.get(url, body="Error", status=500)
+            rsp4 = responses.get(url, body="OK", status=200)
+
+            session = requests.Session()
+
+            adapter = requests.adapters.HTTPAdapter(
+                max_retries=Retry(
+                    total=4,
+                    backoff_factor=0.1,
+                    status_forcelist=[500],
+                    method_whitelist=["GET", "POST", "PATCH"],
+                )
+            )
+            session.mount("https://", adapter)
+
+            resp = session.get(url)
+
+            assert resp.status_code == 200
+            assert rsp1.call_count == 1
+            assert rsp2.call_count == 1
+            assert rsp3.call_count == 1
+            assert rsp4.call_count == 1
+
+        run()
+        assert_reset()
+
+    @pytest.mark.parametrize("raise_on_status", (True, False))
+    def test_max_retries_exceed(self, raise_on_status):
+        @responses.activate(registry=registries.OrderedRegistry)
+        def run():
+            url = "https://example.com"
+            rsp1 = responses.get(url, body="Error", status=500)
+            rsp2 = responses.get(url, body="Error", status=500)
+            rsp3 = responses.get(url, body="Error", status=500)
+
+            session = requests.Session()
+
+            adapter = requests.adapters.HTTPAdapter(
+                max_retries=Retry(
+                    total=2,
+                    backoff_factor=0.1,
+                    status_forcelist=[500],
+                    method_whitelist=["GET", "POST", "PATCH"],
+                    raise_on_status=raise_on_status,
+                )
+            )
+            session.mount("https://", adapter)
+
+            if raise_on_status:
+                with pytest.raises(MaxRetryError):
+                    session.get(url)
+            else:
+                resp = session.get(url)
+                assert resp.status_code == 500
+
+            assert rsp1.call_count == 1
+            assert rsp2.call_count == 1
+            assert rsp3.call_count == 1
+
+        run()
+        assert_reset()
+
+    def test_adapter_retry_untouched(self):
+        """Validate that every new request uses brand-new Retry object"""
+
+        @responses.activate(registry=registries.OrderedRegistry)
+        def run():
+            url = "https://example.com"
+            rsp1 = responses.get(url, body="Error", status=500)
+            responses.add(rsp1)
+            responses.add(rsp1)
+            rsp4 = responses.get(url, body="OK", status=200)
+
+            responses.add(rsp1)
+            responses.add(rsp1)
+            responses.add(rsp1)
+            responses.add(rsp4)
```

Comment:
```suggestion
            error_rsp = responses.get(url, body="Error", status=500)
            responses.add(error_rsp)
            responses.add(error_rsp)
            ok_rsp = responses.get(url, body="OK", status=200)

            responses.add(error_rsp)
            responses.add(error_rsp)
            responses.add(error_rsp)
            responses.add(error_rsp)
```

I think these names might be easier for future us when we're re-reading this test.

Reply:
indeed
even here you were tricked, last one should be OK, let me refactor